typing/typemod.ml

```diff
@@ -844,11 +944,25 @@ and transl_signature env sg =
             mksig (Tsig_modtype mtd) env loc :: trem,
             sg :: rem,
             final_env
-        | Psig_open sod ->
-            let (_path, newenv, od) = type_open env sod in
+        | Psig_open sod -> begin
+            let (newenv, od) = type_open env sod in
             let (trem, rem, final_env) = transl_sig newenv srem in
+            let remr = ref rem in
+            (match extract_open od with
+             | None -> ()
+             | Some (_, _, id, _) -> begin
+                 let s_rem = Mty_signature rem in begin
+                 match Mtype.nondep_supertype newenv id s_rem with
+                 | Mty_signature rem' -> remr := rem'
+                 | exception Not_found ->
+                     raise(Error(sod.popen_loc, env,
+                                 Cannot_eliminate_anon_module(id, rem)))
+                 | _ -> assert false
+                 end
+            end);
             mksig (Tsig_open od) env loc :: trem,
-            rem, final_env
+            !remr, final_env
```

Comment:
Can trem contains reference to the generated identifier?  It could confuse external tools that would see references to unbound module identifiers.  What about keeping the generated identifier in the typedtree?